### PR TITLE
513 compatibility and testing

### PR DESCRIPTION
**Pull request by ninjanomnom**
This does all the code changes to allow 513 testing, once this all seems about correct it should be merged even if the servers aren't using 513.

All changes will be made in a backwards compatible way so that this whole process should be harmless.

### CODE REVIEW

**Comment by Jalleo** [comment]
islist() is now, finally, a built-in proc.

**Comment by ninjanomnom** [comment]
@AnturK I'm holding off on fixing the double encodes until we need to or we move completely to 513. They seem harmless to have in 513 and it keeps compatibility with 512

**Comment by actioninja** [comment]
@Jalleo this pr seems like more of a raw compatibiltiy pass than any kind of significant migration.

File: code/__HELPERS/_lists.dm [hunk 1]
```diff
@@ -519,7 +519,7 @@
 		used_key_list[input_key] = 1
 	return input_key
 
-#if DM_VERSION > 512
+#if DM_VERSION > 513
 #error Remie said that lummox was adding a way to get a lists
 #error contents via list.values, if that is true remove this
 #error otherwise, update the version and bug lummox
```
Comment (JJRcop) [inline code/__HELPERS/_lists.dm]: don't forget this part

**Comment by MrPerson** [comment]
God that alpha masking is so much better than the previous solutions.

**Review by SpaceManiac — APPROVED** [review]

**Review by covertcorvid — COMMENTED** [review]
Will removing border from filter break anything for previous versions? I believe 512 is still our "standard"

**Comment by SpaceManiac** [comment]
@covertcorvid 
>Will removing border from filter break anything for previous versions? I believe 512 is still our "standard"

As far as I know, the `border` option did not do anything and was previously being accepted by mistake.

**Pull request merged**